contracts/test/bridge/tokenBridge.ts

```diff
@@ -0,0 +1,184 @@
+import { TIMEOUT } from '../constants'
+import { ethers, starknet, network } from 'hardhat'
+import { Contract } from 'ethers'
+import { uint256, number } from 'starknet'
+import { StarknetContract, HttpNetworkConfig, Account } from 'hardhat/types'
+import { expect } from 'chai'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { expectAddressEquality, loadStarkgateContract, loadProxyContract } from '../utils'
+
+describe('Test starkgate bridge with link token', function () {
+  this.timeout(TIMEOUT)
+  const networkUrl: string = (network.config as HttpNetworkConfig).url
+  let owner: Account
+  let tokenBridgeContract: StarknetContract
+  let linkTokenContract: StarknetContract
+  let deployer: SignerWithAddress
+
+  let starkNetERC20Bridge: Contract
+  let mockStarknetMessaging: Contract
+  let proxy: Contract
+  let testERC20: Contract
+  let newStarkNetERC20Bridge: Contract
+
+  before(async () => {
+    owner = await starknet.deployAccount('OpenZeppelin')
+
+    let tokenBridgeFactory = await starknet.getContractFactory(
+      '../../node_modules/@chainlink-dev/starkgate-contracts/artifacts/token_bridge.cairo',
+    )
+    tokenBridgeContract = await tokenBridgeFactory.deploy({
+      governor_address: owner.starknetContract.address,
+    })
+
+    let linkTokenFactory = await starknet.getContractFactory('link_token')
+    linkTokenContract = await linkTokenFactory.deploy({ owner: tokenBridgeContract.address })
+
+    const accounts = await ethers.getSigners()
+    deployer = accounts[0]
+
+    const contract = await loadStarkgateContract('StarknetERC20Bridge')
+    const starkNetERC20BridgeFactory = new ethers.ContractFactory(
+      contract.abi,
+      contract.bytecode,
+      deployer,
+    )
+    starkNetERC20Bridge = await starkNetERC20BridgeFactory.deploy()
+    await starkNetERC20Bridge.deployed()
+
+    const mockStarknetMessagingFactory = await ethers.getContractFactory(
+      'MockStarkNetMessaging',
+      deployer,
+    )
+    mockStarknetMessaging = await mockStarknetMessagingFactory.deploy()
+    await mockStarknetMessaging.deployed()
+
+    await starknet.devnet.loadL1MessagingContract(networkUrl, mockStarknetMessaging.address)
+
+    const testERC20Factory = await ethers.getContractFactory('TestERC20', deployer)
```

Comment:
I'd like to see just the standard OZ ERC20 token here. 
I don't think a custom ERC20 is required or desired, as we want to test with the default one.

Reply:
The default one doesn't allow us to mint some token during constructor. That's why Starkgate has the TestERC20.sol to be able to set balance for the bridge. I think we can use this contract from OpenZeppelin https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/presets/ERC20PresetFixedSupply.sol